From d2500a0c0e73d4387cde9185edcdf397f52e428b Mon Sep 17 00:00:00 2001
From: "Martin K. Petersen" <martin.petersen@oracle.com>
Date: Sat, 3 Jul 2021 11:58:33 -0400
Subject: [PATCH] scsi: blkcg: Fix application ID config options
Git-commit: d2500a0c0e73d4387cde9185edcdf397f52e428b
Patch-mainline: v5.14-rc1
References: git-fixes
Modified-by-SEL: No


Commit d2bcbeab4200 ("scsi: blkcg: Add app identifier support for
blkcg") introduced an FC_APPID config option under SCSI. However, the
added config option is not used anywhere. Simply remove it.

The block layer BLK_CGROUP_FC_APPID config option is what actually
controls whether the application ID code should be built or not. Make
this option dependent on NVMe over FC since that is currently the only
transport which supports the capability.

Fixes: d2bcbeab4200 ("scsi: blkcg: Add app identifier support for blkcg")
Reported-by: Linus Torvalds <torvalds@linux-foundation.org>
Signed-off-by: Martin K. Petersen <martin.petersen@oracle.com>
Reviewed-by: Christoph Hellwig <hch@lst.de>
Signed-off-by: Linus Torvalds <torvalds@linux-foundation.org>
Signed-off-by: Guoqing Jiang <guoqing.jiang@suse.com>
---
 block/Kconfig        |  2 +-
 drivers/scsi/Kconfig | 13 -------------
 2 files changed, 1 insertion(+), 14 deletions(-)

diff --git a/block/Kconfig b/block/Kconfig
index 7885a27bd949..fd732aede922 100644
--- a/block/Kconfig
+++ b/block/Kconfig
@@ -153,7 +153,7 @@ config BLK_CGROUP_IOLATENCY
 
 config BLK_CGROUP_FC_APPID
 	bool "Enable support to track FC I/O Traffic across cgroup applications"
-	depends on BLK_CGROUP=y
+	depends on BLK_CGROUP && NVME_FC
 	help
 	  Enabling this option enables the support to track FC I/O traffic across
 	  cgroup applications. It enables the Fabric and the storage targets to
diff --git a/drivers/scsi/Kconfig b/drivers/scsi/Kconfig
index 4dc42a8ff71a..8f44d433e06e 100644
--- a/drivers/scsi/Kconfig
+++ b/drivers/scsi/Kconfig
@@ -235,19 +235,6 @@ config SCSI_FC_ATTRS
 	  each attached FiberChannel device to sysfs, say Y.
 	  Otherwise, say N.
 
-config FC_APPID
-	bool "Enable support to track FC I/O Traffic"
-	depends on BLOCK && BLK_CGROUP
-	depends on SCSI
-	select BLK_CGROUP_FC_APPID
-	default y
-	help
-	  If you say Y here, it enables the support to track
-	  FC I/O traffic over fabric. It enables the Fabric and the
-	  storage targets to identify, monitor, and handle FC traffic
-	  based on VM tags by inserting application specific
-	  identification into the FC frame.
-
 config SCSI_ISCSI_ATTRS
 	tristate "iSCSI Transport Attributes"
 	depends on SCSI && NET
-- 
2.26.2

